From 93c40564b94367c6ce072d66479af58afa3f08e0 Mon Sep 17 00:00:00 2001
From: Tomi Valkeinen <tomi.valkeinen@ideasonboard.com>
Date: Fri, 29 Sep 2023 17:14:31 +0300
Subject: [PATCH] media: rp1: cfe: Add missing compressed remaps

16-bit bayer formats are missing compressed remap definitions. Add them.

Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ideasonboard.com>
---
 drivers/media/platform/raspberrypi/rp1_cfe/cfe_fmts.h | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/media/platform/raspberrypi/rp1_cfe/cfe_fmts.h
+++ b/drivers/media/platform/raspberrypi/rp1_cfe/cfe_fmts.h
@@ -197,24 +197,28 @@ static const struct cfe_fmt formats[] =
 		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
 		.depth = 16,
 		.flags = CFE_FORMAT_FLAG_FE_OUT,
+		.remap = { V4L2_PIX_FMT_SBGGR16, V4L2_PIX_FMT_PISP_COMP1_BGGR },
 	},
 	{
 		.fourcc = V4L2_PIX_FMT_SGBRG16,
 		.code = MEDIA_BUS_FMT_SGBRG16_1X16,
 		.depth = 16,
 		.flags = CFE_FORMAT_FLAG_FE_OUT,
+		.remap = { V4L2_PIX_FMT_SGBRG16, V4L2_PIX_FMT_PISP_COMP1_GBRG },
 	},
 	{
 		.fourcc = V4L2_PIX_FMT_SGRBG16,
 		.code = MEDIA_BUS_FMT_SGRBG16_1X16,
 		.depth = 16,
 		.flags = CFE_FORMAT_FLAG_FE_OUT,
+		.remap = { V4L2_PIX_FMT_SGRBG16, V4L2_PIX_FMT_PISP_COMP1_GRBG },
 	},
 	{
 		.fourcc = V4L2_PIX_FMT_SRGGB16,
 		.code = MEDIA_BUS_FMT_SRGGB16_1X16,
 		.depth = 16,
 		.flags = CFE_FORMAT_FLAG_FE_OUT,
+		.remap = { V4L2_PIX_FMT_SRGGB16, V4L2_PIX_FMT_PISP_COMP1_RGGB },
 	},
 	/* PiSP Compressed Mode 1 */
 	{
